operator/deploy/crd.yaml

```diff
@@ -30,11 +30,335 @@ spec:
+                            insecure_skip_verify:
+                              type: boolean
+                            password:
+                              type: string
+                            token:
+                              type: string
+                            type:
+                              type: string
+                            use_kiali_token:
+                              type: boolean
+                            username:
+                              type: string
+                        custom_metrics_url:
+                          type: string
+                        url:
+                          type: string
+                    tracing:
+                      type: object
+                      properties:
+                        auth:
+                          type: object
+                          properties:
+                            ca_file:
+                              type: string
+                            insecure_skip_verify:
+                              type: boolean
+                            password:
+                              type: string
+                            token:
+                              type: string
+                            type:
+                              type: string
+                            use_kiali_token:
+                              type: boolean
+                            username:
+                              type: string
+                        enabled:
+                          type: boolean
+                        in_cluster_url:
+                          type: string
+                        namespace:
+                          type: string
+                        port:
+                          type: number
+                          minimum: 1
+                          maximum: 65535
+                        service:
+                          type: string
+                        url:
+                          type: string
+                identity:
+                  type: object
+                  properties:
+                    cert_file:
+                      type: string
+                    private_key_file:
+                      type: string
+                istio_labels:
+                  type: object
+                  properties:
+                    app_label_name:
+                      type: string
+                    version_label_name:
+                      type: string
+                kubernetes_config:
+                  type: object
+                  properties:
+                    burst:
+                      type: number
+                    cache_duration:
+                      type: number
+                    cache_enabled:
+                      type: boolean
+                    cache_istio_types:
+                      type: array
+                      items:
+                        type: string
+                        enum:
```

Comment:
cache_istio_types is not an enum - it is a list of strings - similar to the "excluded_workloads" below. The values here under "enum" is the default value.

Reply:
The comment in 
https://github.com/kiali/kiali/blob/9dda43d4bc5bac06e1399da8238cc9bfb83ecc82/operator/deploy/kiali/kiali_cr.yaml#L443-L444

gives the idea that these CRDs are the only supported ones. So, I though it was "good" to put the enum :/
By removing this, perhaps that comment should be fixed.

Reply:
Oh, I get it now. I didn't notice this is a list of "items" of enum type.

That said, I think we should not use enum here and just use string. This will allow for future expansion of the cache WITHOUT requiring us to update the version of the CRD.

Remember, now that we have a schema, we are now required to update the version of the CRD each and every time something is added (and it isn't a trivial thing to bump up the CRD version - people will have created Kiali CRs that needs to eventually be modified, Service Mesh operator will be using a specific version of the Kiali CRD when it writes out the Kiali CR, upstream Istio will eventually be using this, and [OLM has rules about CRD updates](https://github.com/operator-framework/operator-lifecycle-manager/blob/0.12.0/doc/design/dependency-resolution.md#customresourcedefinition-crd-upgrade)). It will be a pain to have to bump the CRD version simply because we added a new type that can be cached. That really shouldn't involve a new CRD version being created.

Reply:
Hmm I see and I agree. OK, I'll remove the enum.
This makes me think that also the `strategy` shouldn't have the enum.

Reply:
> This makes me think that also the `strategy` shouldn't have the enum.

That is a good point.